src/sass/parse.js

```diff
@@ -3239,55 +3239,80 @@ function checkParentSelector(i) {
 
 /**
  * Get node with a parent selector
- * @return {Array} `['parentSelector']`
+ * @return {Node}
  */
 function getParentSelector() {
-  var startPos = pos++;
+  const startPos = pos;
+  const token = tokens[startPos];
+
+  pos++;
 
-  var token = tokens[startPos];
   return newNode(NodeType.ParentSelectorType, '&', token.ln, token.col);
 }
 
+/**
+ * Check if token is a parent selector extension, e.g. `&--foo-bar`
+ * @param {number} i Token's index number
+ * @returns {number} Length of the parent selector extension
+ */
 function checkParentSelectorExtension(i) {
-  if (i >= tokensLength) return 0;
-
-  let start = i;
+  const start = i;
   let l;
 
+  if (i >= tokensLength) return 0;
+
   while (i < tokensLength) {
     if (l = checkNumber(i) || checkIdentOrInterpolation(i)) i += l;
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
     else break;
   }
 
   return i - start;
 }
 
+/**
+ * Get parent selector extension node
+ * @return {Node}
+ */
 function getParentSelectorExtension() {
-  let type = NodeType.ParentSelectorExtensionType;
-  let token = tokens[pos];
-  let line = token.ln;
-  let column = token.col;
+  const type = NodeType.ParentSelectorExtensionType;
+  const token = tokens[pos];
+  const line = token.ln;
+  const column = token.col;
   let content = [];
 
   while (pos < tokensLength) {
     if (checkNumber(pos)) {
       content.push(getNumber());
     } else if (checkIdentOrInterpolation(pos)) {
       content = content.concat(getIdentOrInterpolation());
-    } else {
-      break;
-    }
+    } else if (tokens[pos].type === TokenType.HyphenMinus) {
```

Comment:
Here you propose that in `&----` there will be four `ident` nodes with content `-` in each. Should not it be just one node with content `----`?